Dockerfile

```
@@ -0,0 +1,47 @@
+# This builds Boulder in a Docker container, then creates an image
+# containing just the built Boulder binaries plus some ancillary
+# files that are useful for predeployment testing.
+FROM docker.io/ubuntu:24.04 AS builder
+
+ARG COMMIT_ID
+ARG GO_VERSION
+ARG VERSION
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get --assume-yes --no-install-recommends --update install \
+    ca-certificates curl gcc git gnupg2 libc6-dev
+
+COPY tools/fetch-and-verify-go.sh /tmp
+RUN /tmp/fetch-and-verify-go.sh ${GO_VERSION}
+RUN tar -C /opt -xzf go.tar.gz
```

Combining two of these layers and deleting the gzipped tarball will keep the layers and the overall image smaller.

```suggestion
COPY tools/fetch-and-verify-go.sh /tmp/
RUN /tmp/fetch-and-verify-go.sh "${GO_VERSION}" && \
    tar -C /opt -xzf go.tar.gz && rm -f go.tar.gz
```

---

Same as above; we get to skip some of these optimizations and have more straightforward code by using the builder pattern.